### PR TITLE
MergeController and test fixtures

### PR DESCRIPTION
**Pull request by cjduffett**
Added the `MergeController` struct to pass the mongodb connection and FHIR host to the various routes. Added a mock FHIR server and mongodb server to `server_test.go`, as well as 2 patient bundles that get inserted into the FHIR database during `SetupSuite()`.

The server test suite builds off of the `MongoSuite` testing utility from `intervention-engine/ie/testutil`, and uses patient bundle fixtures from `intervention-engine/fhir`. These fixtures may or may not meet our testing needs, but they're something to start with.

I also updated our dependencies with Glide. Most notably, I fixed the `intervention-engine/fhir` dependency on the `stu3_aug2016` branch (FHIR 1.6). This may need to be changed soon if the FHIR server gets updated to FHIR 1.8.

### CODE REVIEW

File: server/merge_controller.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		session:  session,
+		fhirHost: fhirHost,
+	}
+}
+
+// Merges 2 FHIR bundles of patient resources given the URLs to both bundles.
+func (m *MergeController) merge(c *gin.Context) {
+	source1 := c.Query("source1")
+	source2 := c.Query("source2")
+	c.String(http.StatusOK, "Merging records %s and %s", source1, source2)
```
Comment (cmoesel) [inline server/merge_controller.go]: I'm going to assume that the fact that this doesn't actually do any merging is intentional.  And the same for the rest of the functions in this file.

File: server/server.go [hunk 2]
```diff
@@ -50,7 +50,11 @@ func (p *PTMergeServer) Run() {
 	}
 	log.Printf("Connected to mongodb at %s\n", p.DatabaseHost)
 
-	p.Database = session.DB(p.DatabaseName)
+	// this master database session should be copied using session.Copy()
+	// before making requests to the database. This protects the connection
+	// to mongo if for any reason a database operation times out.
```
Comment (cmoesel) [inline server/server.go]: You say the `session` should be copied, but... I don't see it copied in `RegisterRoutes` or `MergeController`, but maybe that's because they don't actually _use_ it (they only assign it)?

File: server/server_test.go [hunk 3]
```diff
@@ -1,94 +1,148 @@
...
+	// See testutil/mongo_suite.go for more.
+	s.TearDownDBServer()
 }
 
 func (s *ServerTestSuite) TestInitiateMerge() {
-	source1 := "http://www.example.com/fhir/Patient/12345"
-	source2 := "http://www.example.com/fhir/Patient/67890"
-	req := s.Server.URL + "/merge?source1=" + url.QueryEscape(source1) + "&source2=" + url.QueryEscape(source2)
+	source1 := s.FHIRServer.URL + "/Patient?_id=" + s.patientIDs[0] + "&_include=*&_revinclude=*"
+	source2 := source1
```
Comment (cmoesel) [inline server/server_test.go]: No biggie, but this test would be more meaningful if source1 and source2 were different.  As it is now, if the server mistakenly took source1 twice (instead of source1 and source2) this test would still pass.

File: server/server_test.go [hunk 4]
```diff
@@ -1,94 +1,148 @@
...
 
 type ServerTestSuite struct {
-	suite.Suite
-	Engine *gin.Engine
-	Server *httptest.Server
+	// The MongoSuite from IE provides useful features for starting/stopping
+	// a test mongo database, as well as inserting test fixtures.
+	testutil.MongoSuite
+	PTMergeServer *httptest.Server
+	FHIRServer    *httptest.Server
```
Comment (cmoesel) [inline server/server_test.go]: You've integrated the FHIRServer into the tests, but I don't think we can really know if it works because the PtMatch code never actually queries the FHIR server, right?  I assume you're just setting this up in anticipation of future capabilities?

Comment (cjduffett) [inline server/server_test.go]: Correct.

File: server/server.go (hunk 2 above)
Comment (cjduffett) [inline server/server.go]: Correct. Eventually it will get passed to the `Merger` and used during the merge process. At that time I expect the session to be copied.

File: server/merge_controller.go (hunk 1 above)
Comment (cjduffett) [inline server/merge_controller.go]: Correct. the Merger is just stubbed out right now.

File: server/server_test.go (hunk 3 above)
Comment (cjduffett) [inline server/server_test.go]: Fair point. I just used the same URL since the `Merger` is only stubbed out right now. In practice I plan to add additional tests not just for a valid merge, but also for all the corner cases where things could go wrong.
